src/helpers.jl

```diff
@@ -38,4 +41,24 @@ end
 #         nwalkers /= 1 # make it floating point
 #     end
 #     dv = DVec(Dict(address=>nwalkers), capacity)
-#     StochasticStyle(::Type{typeof(dv)}) = Style(1.0)
\ No newline at end of file
+#     StochasticStyle(::Type{typeof(dv)}) = Style(1.0)
+
+
+"""
+    walkernumber(w)
+Compute the number of walkers in `w`. In most cases this is identical to
+`norm(w,1)`. For coefficient vectors with
+`StochasticStyle(w) == IsStochastic2Pop` it reports the one norm
+separately for the real and the imaginary part as a `ComplexF64`.
+"""
+walkernumber(w) = norm(w,1) # generic fallback
+# use StochasticStyle trait for dispatch
+walkernumber(w::AbstractDVec) = walkernumber(StochasticStyle(w), w)
+walkernumber(::StochasticStyle, w) = norm(w,1)
+# for AbstractDVec with complex walkers
+function walkernumber(::T, w) where T <: Union{IsStochastic2Pop,
+                                               IsStochastic2PopInitiator,
+                                               IsStochastic2PopWithThreshold
+                                               }
+    return isempty(w) ? 0.0+0.0im : mapreduce(p->abs(real(p)) + abs(imag(p))*im, +, w)|>ComplexF64
```

Comment:
You could replace `mapreduce` with `sum`.

Reply:
done